### PR TITLE
feat(formula): add some information and date function

### LINKED ISSUE #2763: [Feature] More Info Formulas
### Initial checklist

- [X] Is this really a problem?
- [X] I have searched the Github Issues for similar issues, but did not find anything.

### Problem

[How to contribute formulas](https://univer.ai/guides/sheet/advanced/custom-formula#how-to-add-formulas-in-formula-engine)

#### Info formulas
- [x] CELL
- [x] ERROR.TYPE
- [x] TYPE
#### Date formulas
- [x] DAYS
- [x] DAYS360
- [x] EOMONTH
- [x] WEEKNUM
- [x] ISOWEEKNUM
- [x] YEARFRAC

### PR DESCRIPTION
**Pull request by wpxp123456**
close #2763

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/engine-formula/src/functions/date/weeknum/index.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            if (returnType.isBoolean()) {
+                return ErrorValueObject.create(ErrorType.VALUE);
+            }
+
+            if (returnType.isString() && !isRealNum(returnTypeValue)) {
+                return ErrorValueObject.create(ErrorType.VALUE);
+            }
+        }
+
+        const returnTypeMap: {
```
Comment (Dushusir) [inline packages/engine-formula/src/functions/date/weeknum/index.ts:85]: 常量可以放到函数外面，防止每次创建

**Review by Dushusir — APPROVED** [review]

**Comment by oumomomo** [comment]
单元格内输入=TYPE(2+A1)，A1=Smith，计算结果有误
image

File: packages/engine-formula/src/functions/date/weeknum/index.ts (hunk 1 above)
Comment (wpxp123456) [reply packages/engine-formula/src/functions/date/weeknum/index.ts:85]: ok

**Comment by oumomomo** [comment]
输入=YEARFRAC(A2,A3,A4),错误码不一致
企业微信截图_17214604803711

**Comment by oumomomo** [comment]
输入=YEARFRAC(A3,A4,0)，计算结果不一致，A3=1/1/2012  A4=7/30/2012
企业微信截图_17214614838812

**Pull request merged**